### PR TITLE
Feat: add pending phase in workflow step

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

This PR adds pending phase for workflow step, for example, if the step is executed but waiting for its inputs or depends on other steps, now its phase is `pending` and message is `Pending on Inputs/DependsOn: xxx`.

Previously, if the step is executed but waiting for its inputs, the step will have no status, and the backoff time will not work.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: apis/core.oam.dev/common/types.go [hunk 1]
```diff
@@ -420,6 +420,8 @@ const (
 	WorkflowStepPhaseStopped WorkflowStepPhase = "stopped"
 	// WorkflowStepPhaseRunning will make the controller continue the workflow.
 	WorkflowStepPhaseRunning WorkflowStepPhase = "running"
+	// WorkflowStepPhasePending will make the controller wait for the step to run.
+	WorkflowStepPhasePending WorkflowStepPhase = "pending"
```
Comment (barnettZQG) [inline apis/core.oam.dev/common/types.go:424]: what conditions is the status pending?

Comment (FogDong) [reply apis/core.oam.dev/common/types.go:424]: For example, the step is waiting for the inputs or dependsOn

**Review by Somefive — APPROVED** [review]

**Review by wonderflow — APPROVED** [review]

**Comment by wonderflow** [comment]
Does it fix  https://github.com/kubevela/kubevela/issues/4188 ？

Please add some detail description in the PR content

**Review by leejanee — APPROVED** [review]
LGTM

**Pull request merged**